planner/core/exhaust_physical_plans.go

```diff
@@ -2765,16 +2765,18 @@ func (lw *LogicalWindow) tryToGetMppWindows(prop *property.PhysicalProperty) []P
 			return nil
 		}
 		if lw.Frame != nil && lw.Frame.Type == ast.Ranges {
-			if _, err := expression.ExpressionsToPBList(lw.SCtx().GetSessionVars().StmtCtx, lw.Frame.Start.CalcFuncs, lw.ctx.GetClient()); err != nil {
-				lw.SCtx().GetSessionVars().RaiseWarningWhenMPPEnforced(
-					"MPP mode may be blocked because window function frame can't be pushed down, because " + err.Error())
-				return nil
-			}
-			if _, err := expression.ExpressionsToPBList(lw.SCtx().GetSessionVars().StmtCtx, lw.Frame.End.CalcFuncs, lw.ctx.GetClient()); err != nil {
-				lw.SCtx().GetSessionVars().RaiseWarningWhenMPPEnforced(
-					"MPP mode may be blocked because window function frame can't be pushed down, because " + err.Error())
-				return nil
-			}
+			// TODO uncomment these codes after tiflash support range frame type
```

Comment:
Why comment these codes? I think we should keep the warnings if it can't be pushed to TiFlash

Reply:
> Why comment these codes? I think we should keep the warnings if it can't be pushed to TiFlash

okk, i will uncomment them